mkosi

```diff
@@ -2250,92 +2378,35 @@ def install_arch(args: CommandLineArguments, root: str, do_run_build_script: boo
         if args.encrypt:
             packages.add("cryptsetup")
             packages.add("device-mapper")
-        if not kernel_packages:
-            # No user-specified kernel
-            kernel_packages.add("linux")
         if args.bios_partno:
             packages.add("grub")
 
-        packages.add("mkinitcpio")
-
-    def run_pacstrap(packages: Set[str]) -> None:
-        cmdline = ["pacstrap", "-C", pacman_conf, "-GM", root]
-        # pacstrap handles gpg-agent on its own
-        run(cmdline + list(packages), check=True)
+        packages.add("dracut")
+        packages.add("binutils")
 
-    # Set up system with packages from the base group
-    run_pacstrap(packages)
+    packages.update(args.packages)
 
-    if not do_run_build_script and args.bootable:
-        # Patch mkinitcpio configuration so:
-        # 1) we remove autodetect and
-        # 2) we add the modules needed for encrypt.
-        def jj(line: str) -> str:
-            if line.startswith("HOOKS="):
-                if args.encrypt == "all":
-                    return 'HOOKS="systemd modconf block sd-encrypt filesystems keyboard fsck"\n'
-                else:
-                    return 'HOOKS="systemd modconf block filesystems fsck"\n'
-            return line
-        patch_file(os.path.join(root, "etc/mkinitcpio.conf"), jj)
-
-        # Disable fallback images to save time. We override the preset files
-        # before installing any kernels to have the updated config available
-        # when the mkinitcpio pacman hook is run during installation.
-        presetdir = os.path.join(root, "etc/mkinitcpio.d")
-        for kernel in kernel_packages:
-            preset = os.path.join(presetdir, f"{kernel}.preset")
-            with open(preset, "w") as f:
-                f.write(
-                    dedent(
-                        f"""
-                        ALL_config="/etc/mkinitcpio.conf"
-                        ALL_kver="/boot/vmlinuz-{kernel}"
-                        PRESETS=('default')
-                        default_image="/boot/initramfs-{kernel}.img"
-                        """
-                    )
-                )
+    official_kernel_packages = {
+        "linux",
+        "linux-lts",
+        "linux-hardened",
+        "linux-zen",
+    }
 
-    # Install the user-specified packages and kernel
-    packages = set(args.packages)
-    if not do_run_build_script and args.bootable:
-        packages |= kernel_packages
+    has_kernel_package = any(package in official_kernel_packages for package in args.packages)
```

Comment:
for the usage below this can be simplified to
```suggestion
    has_kernel_package = official_kernel_packages.intersection(args.packages)
```

Reply:
I had no idea lists/sets could evaluate to bool. Nice!

Reply:
Every class that implements `__bool__` can be used in boolean contextes. All the standard implement it such, that they are False, when they are empty and True, when they are not. That's also why the empty string is False (also the empty tuple).